src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -625,9 +625,9 @@ sl_status_t show_scan_results(sl_wifi_scan_result_t * scan_result)
                                    (char *) scan_result->scan_info[idx].ssid); // +1 for null termination
 
         // if user has provided ssid, then check if the current scan result ssid matches the user provided ssid
-        if (wfx_rsi.scan_ssid != nullptr &&
-            (strncmp(wfx_rsi.scan_ssid, cur_scan_result.ssid, std::min(strlen(wfx_rsi.scan_ssid), strlen(cur_scan_result.ssid))) ==
-             0))
+        if (wfx_rsi.scan_ssid != nullptr && 
+            (strlen(wfx_rsi.scan_ssid) != strlen(cur_scan_result.ssid) || 
+             strncmp(wfx_rsi.scan_ssid, cur_scan_result.ssid, strlen(wfx_rsi.scan_ssid)) != 0))
```

Comment:
This is assumed safe if both strings are correctly null terminated. I do see we handle it for cur_scan_result just above. Is wfx_rsi.scan_ssid always null-terminated too?

Reply:
Returned SSIDs for the wiseconnect are always null terminated based on how the sdk manages output ssid (which differs from the input structure).